### ISSUE TITLE
Marker classes errors

### ISSUE THREAD

**Issue by RinyVT** [franzholz/tt_products#16]
Hi,

I'm using the tt_products extension version 2.7.29, but unfortunately it produces errors.
In the class /view/class.tx_ttproducts_catlist_view_base.php there is in initialization of tx_ttproducts_subpartmarker:
```
$this->subpartmarkerObj = t3lib_div::makeInstance('tx_ttproducts_subpartmarker');
$this->subpartmarkerObj->init($pibase->cObj);
```

But the $pibase variable doesn't exist, it should be $this->cObj
When I change that it works, but unfortunately I can't XCLASS the class since it's not been called by makeInstance or getUserObj but is used with "extends" by tx_ttproducts_menucat_view, tx_ttproducts_selectcat_view, tx_ttproducts_catlist_view.

The problem with $pibase->cObj also occurs in class.tx_ttproducts_basket_view.php and class.tx_ttproducts_currency_view.php (in the last class it should be replaced by $this->pibase->cObj)

Btw, the problem that occurs is that in tx_ttproducts_subpartmarker the call to stdWrap on $this->cObj fails.

**Comment by franzholz** [comment]
$pibase exists:
    
    class tx_ttproducts_catlist_view extends tx_ttproducts_catlist_view_base {

        public function init ($pibase, $pid_list, $recursive, $pid) {

**Comment by RinyVT** [comment]
Yes it exists in the tx_ttproducts_catlist_view but not in:
https://github.com/franzholz/tt_products/blob/master/view/class.tx_ttproducts_catlist_view_base.php

You can see there on line 78 that it uses $pibase which is a variable which doesn't exist.

**Comment by franzholz** [comment]
The master branch is for tt_products 2.8.x.
The branch for tt_products 2.7.x. is here:

https://github.com/franzholz/tt_products/blob/version2_7_0/view/class.tx_ttproducts_catlist_view_base.php

**Comment by franzholz** [comment]
The master branch is for tt_products 2.8.x.
The branch for tt_products 2.7.x. is here:

https://github.com/franzholz/tt_products/blob/version2_7_0/view/class.tx_ttproducts_catlist_view_base.php

**Comment by franzholz** [comment]
I have made this bugfix for tt_products 2.8.x:

https://github.com/franzholz/tt_products/commit/a8c219430bc83a031245ae069639946f8b994b2c

**Comment by franzholz** [comment]
I have added a small fix for tt_products 2.7.29. However there is no bug in this version.

https://github.com/franzholz/tt_products/commit/7026f8691e335e15f2be308377eacf32d27900b0
